src/main/java/Deadline.java

```diff
@@ -0,0 +1,63 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates a task with a deadline
+ */
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```

Comment:
The variable name looks fine here with the `LocalDate` type specified right before it, however, it may be less obvious in other parts of the code where the type of `by` isn't immediately obvious. Maybe a more descriptive name here might be better?